mcl/bn.nim

```diff
@@ -0,0 +1,534 @@
+# mcl
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms
+
+{.push raises: [].}
+
+import
+  std/os
+
+const
+  basePath = currentSourcePath.parentDir.parentDir & "/vendor/mcl"
+  headerPath = basePath & "/include"
+  srcPath = basePath & "/src"
+
+const
+  MCL_FP_BIT = 384
+  MCL_FR_BIT = 256
+
+{.passc: "-DMCL_FP_BIT=" & $MCL_FP_BIT.}
+{.passc: "-DMCL_FR_BIT=" & $MCL_FR_BIT.}
+{.passc: "-DCYBOZU_DONT_USE_STRING -DCYBOZU_DONT_USE_EXCEPTION".}
+{.passc: "-I" & headerPath.}
+
+when defined(windows):
+  {.compile: srcPath & "/fp.cpp".}
+else:
+  {.compile(srcPath & "/fp.cpp", "-fno-lto").}
+
+when defined(amd64):
+  when defined(linux):
+    {.compile(srcPath & "/msm_avx.cpp", "-fno-lto -mavx512f -mavx512ifma -std=c++11").}
+    {.hint: "MSM AVX".}
+  when defined(windows):
+    {.compile(srcPath & "/msm_avx.cpp", "-mavx512f -mavx512ifma -std=c++11").}
+  when defined(linux):
+    {.compile: srcPath & "/asm/bint-x64-amd64.S".}
+    {.hint: "LINUX BINT".}
+  when defined(windows):
+    {.compile: srcPath & "/asm/bint-x64-mingw.S".}
+    {.hint: "WINDOWS BINT".}
+  when defined(macosx):
+    {.passc: "-DMCL_BINT_ASM=1 -DMCL_BINT_ASM_X64=0 -DMCL_USE_LLVM".}
+    {.compile: srcPath & "/base64.ll".}
+    {.compile: srcPath & "/bint64.ll".}
+    when not defined(clang):
+      {.error: "requires clang on macos amd64".}
+    {.hint: "MACOS BINT".}
+
+when sizeof(int) == 8 and not defined(amd64):
+  {.passc: "-DMCL_BINT_ASM=1 -DMCL_BINT_ASM_X64=0 -DMCL_USE_LLVM".}
+  {.compile: srcPath & "/base64.ll".}
+  {.compile: srcPath & "/bint64.ll".}
+  when not defined(clang):
+    {.error: "requires clang on " & hostOS & " " & hostCPU.}
```

Comment:
macOS gcc is infact a copy/link to clang post Apple Silicon transition
not sure if we need this check, since macOS intel versions are fully deprecated now